vector_setters.go

```diff
@@ -88,93 +84,105 @@ func setBool[S any](vec *vector, rowIdx C.idx_t, val S) error {
 	return nil
 }
 
-func setTS[S any](vec *vector, rowIdx C.idx_t, val S) error {
-	ts, err := getCTimestamp(vec.Type, val)
+func setTS[S any](vec *vector, rowIdx mapping.IdxT, val S) error {
+	ts, err := getMappedTimestamp(vec.Type, val)
 	if err != nil {
 		return err
 	}
 	setPrimitive(vec, rowIdx, ts)
 	return nil
 }
 
-func setDate[S any](vec *vector, rowIdx C.idx_t, val S) error {
-	date, err := getCDate(val)
+func setDate[S any](vec *vector, rowIdx mapping.IdxT, val S) error {
+	date, err := getMappedDate(val)
 	if err != nil {
 		return err
 	}
 	setPrimitive(vec, rowIdx, date)
 	return nil
 }
 
-func setTime[S any](vec *vector, rowIdx C.idx_t, val S) error {
+func setTime[S any](vec *vector, rowIdx mapping.IdxT, val S) error {
 	ticks, err := getTimeTicks(val)
 	if err != nil {
 		return err
 	}
 
 	switch vec.Type {
 	case TYPE_TIME:
-		var duckTime C.duckdb_time
-		duckTime.micros = C.int64_t(ticks)
-		setPrimitive(vec, rowIdx, duckTime)
+		var ti mapping.Time
+		mapping.TimeSetMicros(&ti, ticks)
+		setPrimitive(vec, rowIdx, ti)
 	case TYPE_TIME_TZ:
 		// The UTC offset is 0.
-		duckTimeTZ := C.duckdb_create_time_tz(C.int64_t(ticks), 0)
-		setPrimitive(vec, rowIdx, duckTimeTZ)
+		ti := mapping.CreateTimeTZ(ticks, 0)
+		setPrimitive(vec, rowIdx, ti)
 	}
 	return nil
 }
 
-func setInterval[S any](vec *vector, rowIdx C.idx_t, val S) error {
-	var interval Interval
+func setInterval[S any](vec *vector, rowIdx mapping.IdxT, val S) error {
+	var i Interval
 	switch v := any(val).(type) {
 	case Interval:
-		interval = v
+		i = v
 	default:
-		return castError(reflect.TypeOf(val).String(), reflect.TypeOf(interval).String())
+		return castError(reflect.TypeOf(val).String(), reflect.TypeOf(i).String())
 	}
-	var interval2 C.duckdb_interval
-	interval2.days = C.int32_t(interval.Days)
-	interval2.months = C.int32_t(interval.Months)
-	interval2.micros = C.int64_t(interval.Micros)
-	setPrimitive(vec, rowIdx, interval2)
+	var interval mapping.Interval
+	mapping.IntervalSetMonths(&interval, i.Months)
+	mapping.IntervalSetDays(&interval, i.Days)
+	mapping.IntervalSetMicros(&interval, i.Micros)
+	setPrimitive(vec, rowIdx, interval)
 	return nil
 }
 
-func setHugeint[S any](vec *vector, rowIdx C.idx_t, val S) error {
+func setHugeint[S any](vec *vector, rowIdx mapping.IdxT, val S) error {
 	var err error
-	var fv C.duckdb_hugeint
+	var fv mapping.HugeInt
 	switch v := any(val).(type) {
 	case uint8:
-		fv = C.duckdb_hugeint{lower: C.uint64_t(v)}
+		mapping.HugeIntSetLower(&fv, uint64(v))
+		mapping.HugeIntSetUpper(&fv, 0)
 	case int8:
```

Comment:
Did you check any benchmarks for this? This is 2 function calls for such a minor thing might be a lot. Go does not inline functions across modules IIRC. Would it be possible to just expose a type like it was before? If you don't feel like doing it now thats ok, can be looked into later after this is merged

Reply:
I mean why not just set upper/lower in the same call? `mapping.HugeIntSet(&fv, uint64(v), 0)`